File: packages/maplibre/ios/Plugin/Classes/Options/AddMarkersOptions.swift
import Capacitor
import Foundation

@objc public class AddMarkersOptions: NSObject {
    let mapId: String
    let markers: [Marker]

    init(_ call: CAPPluginCall) throws {
        guard let markers = Marker.fromJSArray(call.getArray("markers")) else {
            throw CustomError.markersMissing
        }
        self.mapId = try MapLibreHelper.getString(call, "mapId", .mapIdMissing)
        self.markers = markers
    }
}

File: packages/maplibre/android/src/main/java/io/capawesome/capacitorjs/plugins/maplibre/classes/options/AddMarkersOptions.java
package io.capawesome.capacitorjs.plugins.maplibre.classes.options;

import androidx.annotation.NonNull;
import com.getcapacitor.JSArray;
import com.getcapacitor.JSObject;
import com.getcapacitor.PluginCall;
import io.capawesome.capacitorjs.plugins.maplibre.MapLibreHelper;
import io.capawesome.capacitorjs.plugins.maplibre.classes.CustomExceptions;
import io.capawesome.capacitorjs.plugins.maplibre.classes.Marker;
import java.util.ArrayList;
import java.util.List;

public class AddMarkersOptions {

    @NonNull
    private final String mapId;

    @NonNull
    private final List<Marker> markers = new ArrayList<>();

    public AddMarkersOptions(@NonNull PluginCall call) throws Exception {
        JSArray markersArray = call.getArray("markers", null);
        if (markersArray == null) {
            throw CustomExceptions.MARKERS_MISSING;
        }
        this.mapId = MapLibreHelper.getMapIdFromCall(call);
        for (JSObject markerObject : MapLibreHelper.createObjectList(markersArray)) {
            this.markers.add(new Marker(markerObject));
        }
    }

    @NonNull
    public String getMapId() {
        return mapId;
    }

    @NonNull
    public List<Marker> getMarkers() {
        return markers;
    }
}
